### DISCUSSION TITLE
get ws in Subscribers

### DISCUSSION THREAD

**Discussion by nntit** [uNetworking/uWebSockets.js#561] [Q&A]
Hello, i am new to nwebsocket, i tried getUserData and i can get the number of subscribers in a topic by this, but, how can i get userdata in the Subscribers.

Thanks all.

**Comment by hst-m** [comment]
Get number of subscribers with [app.numSubscribers(topic)](https://unetworking.github.io/uWebSockets.js/generated/interfaces/templatedapp.html#numsubscribers), to get a list of subscribers you need to track them yourself

**Comment by nntit** [comment]
so is there anyway that i can get a socket of any connection? like socketio, i can get a socket by using its id.

**Reply by hst-m** [reply]
you set the ID

```javascript
const sockets=new Map()
let id=0

app.ws('/',{
   open:ws=>sockets.set(++id,ws)
})

sockets.get(1)
```
